src/core/client/ui/encapsulation/onShadowRootWidthChange.ts

```diff
@@ -81,7 +82,7 @@ export default function onShadowRootWidthChange(
     if (index > -1) {
       array.splice(index, 1);
       if (array.length === 0) {
-        value.resizeObserver.unobserve(shadowRoot.firstChild! as Element);
+        value.resizeObserver.unobserve(shadowRootFirstChild as Element);
```

Comment:
Crazy how even today, in JS land, we're still basically dealing with a moving (ref)-pointer to an object. Since it changed out from underneath us, we can't clean up properly. Wild.

Good catch! Would be interested to hear how you sussed this out, as it looks like a very innocuous and hidden bug!

Reply:
should we rewrite the frontend in Rust?